### PR TITLE
MGMT-7174 - Add "Copy to clipboard"  button to the Web Console troubleshooting modal

### PR DESCRIPTION
**Pull request by jgyselov**
https://issues.redhat.com/browse/MGMT-7174
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,11 +32,11 @@
     "postswagger": "sed -i -r 's/([a-z1-9]+)_([a-z])/\\1\\U\\2/g' src/api/types.ts && yarn prettier"
   },
   "peerDependencies": {
-    "@patternfly/react-code-editor": "^4.1.25",
-    "@patternfly/react-core": "^4.97.2",
-    "@patternfly/react-icons": "^4.9.2",
-    "@patternfly/react-table": "^4.23.2",
-    "@patternfly/react-tokens": "^4.10.2",
+    "@patternfly/react-code-editor": "^4.2.92",
```
Comment (mareklibra) [inline package.json:35]: We need to verify these changes in context of the uhc-portal project.

File: src/components/ui/PrismCode.tsx [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
   <Highlight {...defaultProps} code={code} language={language} theme={theme}>
-    {({ className, style, tokens, getLineProps, getTokenProps }) => (
-      <Text component={TextVariants.pre} className={className} style={style}>
-        {tokens.map((line, i) => (
-          <div key={i} {...getLineProps({ line, key: i })}>
-            {line.map((token, key) => (
-              <span key={key} {...getTokenProps({ token, key })} />
+    {({ className, style, tokens, getLineProps, getTokenProps }) => {
+      return (
+        <React.Fragment>
```
Comment (mareklibra) [inline src/components/ui/PrismCode.tsx:21]: nitpick: We can use `<>` instead.

File: src/components/ui/PrismCode.tsx [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+  copiable = false,
+}) => (
   <Highlight {...defaultProps} code={code} language={language} theme={theme}>
-    {({ className, style, tokens, getLineProps, getTokenProps }) => (
-      <Text component={TextVariants.pre} className={className} style={style}>
-        {tokens.map((line, i) => (
-          <div key={i} {...getLineProps({ line, key: i })}>
-            {line.map((token, key) => (
-              <span key={key} {...getTokenProps({ token, key })} />
+    {({ className, style, tokens, getLineProps, getTokenProps }) => {
```
Comment (mareklibra) [inline src/components/ui/PrismCode.tsx:19]: nitpick:
```suggestion
    {({ className, style, tokens, getLineProps, getTokenProps }) => (<>FOO</>)
```

(we can skip `return` in the arrow function)

File: src/components/ui/PrismCode.tsx (hunk 2 above)
Comment (jgyselov) [reply src/components/ui/PrismCode.tsx]: Done.

File: src/components/ui/PrismCode.tsx (hunk 3 above)
Comment (jgyselov) [reply src/components/ui/PrismCode.tsx]: Done.

File: package.json (hunk 1 above)
Comment (jkilzi) [reply package.json:35]: +1

File: src/components/ui/PrismCode.tsx [hunk 4]
```diff
@@ -1,27 +1,43 @@
...
+        <Text component={TextVariants.pre} className={className} style={style}>
+          {copiable && (
+            <ClipboardCopy
+              isReadOnly
+              variant={'inline-compact'}
+              onCopy={(e) => clipboardCopyFunc(e, code)}
+              style={{ float: 'right', background: 'inherit' }}
+            >
+              {}
+            </ClipboardCopy>
```
Comment (jkilzi) [inline src/components/ui/PrismCode.tsx:30]: @jgyselov does this work?

```suggestion
            <ClipboardCopy
              isReadOnly
              variant={'inline-compact'}
              onCopy={(e) => clipboardCopyFunc(e, code)}
              style={{ float: 'right', background: 'inherit' }}
              children={null}
            />
```

Comment (jgyselov) [reply src/components/ui/PrismCode.tsx:30]: No. I get "semantic error TS2741: Property 'children' is missing in type ... but required in type ...".

File: package.json (hunk 1 above)
Comment (jgyselov) [reply package.json:35]: Yes, it's exactly that. @jkilzi

Comment (jkilzi) [reply package.json:35]: @jgyselov if you have the time we can make use of a screenshot in the PR's description
@mareklibra it turns out that PF version bumping is really necessary, I agree with you, we need to check we are good in `uhc-portal`

File: src/components/ui/PrismCode.tsx (hunk 4 above)
Comment (jkilzi) [reply src/components/ui/PrismCode.tsx:30]: Bummer... How about this?

```suggestion
            <ClipboardCopy
              isReadOnly
              variant={'inline-compact'}
              onCopy={(e) => clipboardCopyFunc(e, code)}
              style={{ float: 'right', background: 'inherit' }}
              children={null}
            />
```

**Comment by jkilzi** [comment]
@jgyselov please rebase on top of master

File: src/components/ui/PrismCode.tsx (hunk 4 above)
Comment (jgyselov) [reply src/components/ui/PrismCode.tsx:30]: @jkilzi That does compile, but the linter doesn't look happy.

**Comment by jgyselov** [comment]
> @jgyselov please rebase on top of master

@jkilzi I did `git rebase master`. I hope this is what was supposed to happen.

File: src/common/components/ui/PrismCode.tsx [hunk 5]
```diff
@@ -1,27 +1,43 @@
...
+          )}
+          {tokens.map((line, i) => (
+            <div key={i} {...getLineProps({ line, key: i })}>
+              {line.map((token, key) => (
+                <span key={key} {...getTokenProps({ token, key })} />
+              ))}
+            </div>
+          ))}
+        </Text>
+      </>
```
Comment (jkilzi) [inline src/common/components/ui/PrismCode.tsx:40]: The React Fragment is redundant

```suggestion
      <Text component={TextVariants.pre} className={className} style={style}>
        {copiable && (
          <ClipboardCopy
            isReadOnly
            variant={'inline-compact'}
            onCopy={(e) => clipboardCopyFunc(e, code)}
            style={{ float: 'right', background: 'inherit' }}
          >
            {}
          </ClipboardCopy>
        )}
        {tokens.map((line, i) => (
          <div key={i} {...getLineProps({ line, key: i })}>
            {line.map((token, key) => (
              <span key={key} {...getTokenProps({ token, key })} />
            ))}
          </div>
        ))}
      </Text>
```

**Review by jkilzi — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (jkilzi) [reply package.json:35]: @jgyselov please open a MR in uhc-portal similar [to this one](https://github.com/openshift-assisted/assisted-ui/pull/459), you can add the link to it here as well or in the JIRA task for better visibility

**Comment by jgyselov** [comment]
Only merge after [this MR](https://gitlab.cee.redhat.com/service/uhc-portal/-/merge_requests/2705).

**Review by jkilzi — APPROVED** [review]

**Pull request merged**